src/nrniv/nrnbbcore_write.cpp

```diff
@@ -532,6 +532,32 @@ DatumIndices::~DatumIndices() {
   if (ion_index) delete [] ion_index;
 }
 
+/**
+   Copy weights from all coreneuron::NrnThread to NetCon instances.
+   This depends on the CoreNEURON weight order for each thread to be
+   the same as was originally sent from NEURON.  See how that order
+   was constructed in mk_cgs_netcon_info below.
+**/
+
+static void nrnthreads_all_weights_return(std::vector<double*>& weights) {
+  std::vector<int> iw; // index for each thread
```

Comment:
Thanks. I was wondering how pedantic I should be since:
```
std::vector<int> iw(std::size_t(nrn_nthread)); // index for each thread
```
gives several warnings and  a compile error on clang
```
/Users/michaelhines/neuron/fornetcon/src/nrniv/nrnbbcore_write.cpp:543:22: warning: 
      parentheses were disambiguated as a function declaration [-Wvexing-parse]
  std::vector<int> iw(std::size_t(nrn_nthread)); // index for each thread
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/michaelhines/neuron/fornetcon/src/nrniv/nrnbbcore_write.cpp:543:23: note: 
      add a pair of parentheses to declare a variable
  std::vector<int> iw(std::size_t(nrn_nthread)); // index for each thread
                      ^
                      (                       )
/Users/michaelhines/neuron/fornetcon/src/nrniv/nrnbbcore_write.cpp:543:20: warning: 
      'iw' has C-linkage specified, but returns user-defined type
      'std::vector<int>' which is incompatible with C [-Wreturn-type-c-linkage]
  std::vector<int> iw(std::size_t(nrn_nthread)); // index for each thread
                   ^
/Users/michaelhines/neuron/fornetcon/src/nrniv/nrnbbcore_write.cpp:555:37: error: 
      subscript of pointer to function type 'std::vector<int> (std::size_t)'
      (aka 'vector<int> (unsigned long)')
      nc->weight_[i] = weights[ith][iw[ith]++];
                                    ^~
2 warnings and 1 error generated.
```
I see now that to cast it needs to be written
```
std::vector<int> iw((std::size_t(nrn_nthread))); // index for each thread
```
to specify the explicit cast.  Anyway, since you indicate there is nothing wrong with using int instead of size_t
in this situation, I'm using your suggestion.  Compiling this file with -Wall gives 8 warnings but none of them have to do with this line and all are "unused..." warnings except for
```
/Users/michaelhines/neuron/fornetcon/src/nrncvode/netcvode.h:18:1: warning: class 'hoc_Item' was previously declared as a struct; this is valid, but may result in linker errors under the Microsoft C++ ABI [-Wmismatched-tags]
class hoc_Item;
```

Reply:
Thanks. I was wondering how pedantic I should be since:
```
std::vector<int> iw(std::size_t(nrn_nthread)); // index for each thread
```
gives several warnings and  a compile error on clang
```
/Users/michaelhines/neuron/fornetcon/src/nrniv/nrnbbcore_write.cpp:543:22: warning: 
      parentheses were disambiguated as a function declaration [-Wvexing-parse]
  std::vector<int> iw(std::size_t(nrn_nthread)); // index for each thread
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/michaelhines/neuron/fornetcon/src/nrniv/nrnbbcore_write.cpp:543:23: note: 
      add a pair of parentheses to declare a variable
  std::vector<int> iw(std::size_t(nrn_nthread)); // index for each thread
                      ^
                      (                       )
/Users/michaelhines/neuron/fornetcon/src/nrniv/nrnbbcore_write.cpp:543:20: warning: 
      'iw' has C-linkage specified, but returns user-defined type
      'std::vector<int>' which is incompatible with C [-Wreturn-type-c-linkage]
  std::vector<int> iw(std::size_t(nrn_nthread)); // index for each thread
                   ^
/Users/michaelhines/neuron/fornetcon/src/nrniv/nrnbbcore_write.cpp:555:37: error: 
      subscript of pointer to function type 'std::vector<int> (std::size_t)'
      (aka 'vector<int> (unsigned long)')
      nc->weight_[i] = weights[ith][iw[ith]++];
                                    ^~
2 warnings and 1 error generated.
```
I see now that to cast it needs to be written
```
std::vector<int> iw((std::size_t(nrn_nthread))); // index for each thread
```
to specify the explicit cast.  Anyway, since you indicate there is nothing wrong with using int instead of size_t
in this situation, I'm using your suggestion.  Compiling this file with -Wall gives 8 warnings but none of them have to do with this line and all are "unused..." warnings except for
```
/Users/michaelhines/neuron/fornetcon/src/nrncvode/netcvode.h:18:1: warning: class 'hoc_Item' was previously declared as a struct; this is valid, but may result in linker errors under the Microsoft C++ ABI [-Wmismatched-tags]
class hoc_Item;
```

Reply:
I see, a bit of a mix of C with C++ there.
We can use C++ casting here:
```
std::vector<int> iw(static_cast<std::size_t>(nrn_nthread)));
```